### DISCUSSION TITLE
ImportError: dynamic module does not define module export function (PyInit__mgis)

### DISCUSSION THREAD

**Discussion by jeanmichelscherer** [thelfer/MFrontGenericInterfaceSupport#93] [General]
Hi Thomas,

I would like to report on how to solve the above error I ran into when installing mgis, because others might face the same issue.

I installed boost version 1.65 with `apt-get install libboost-python1.65-dev` and then ran `cmake .. -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=~/MFrontGenericInterfaceSupport/install-python-3.6/ -Denable-c-bindings=ON -Denable-fortran-bindings=ON -Denable-python-bindings=ON -DPython_ADDITIONAL_VERSIONS=3.6` as recommended in `INSTALL-cmake.md`.

After setting my `$PYTHONPATH` and `$LD_LIBRARY_PATH` adequately, the following error occurred:
```
fenics@9337a8ec9d52:~$ python3
Python 3.6.7 (default, Oct 22 2018, 11:32:17) 
[GCC 8.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import mgis
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/fenics/MFrontGenericInterfaceSupport/install-python-3.6/lib/python3.6/site-packages/mgis/__init__.py", line 1, in <module>
    from ._mgis import *
ImportError: dynamic module does not define module export function (PyInit__mgis)
```
Trying to fix this issue, I came across this thread:
https://sourceforge.net/p/tfel/discussion/installation/thread/32dd128c2a/

Adding `-Denable-broken-boost-python-module-visibility-handling=ON` to the `cmake` command above solved the problem. 
`mgis` is now imported nicely.

Hope this can help future users.

Best,
Jean-Michel

**Comment by thelfer** [comment]
Thanks for the feed-back @jeanmichelscherer. I would admit that I totally forgot about this issue... and the fix !
